module/x/gravity/abci.go

```diff
@@ -366,10 +348,10 @@ func outgoingTxSlashing(ctx sdk.Context, k keeper.Keeper) {
 		// Slash unbonding validators who didn't sign new signer set txs only
 		if sstx, ok := otx.(*types.SignerSetTx); ok {
 			sstxHeight := int64(sstx.Height)
-			for _, validator := range unbondingValidators {
-				signingStartHeight, signingInfoExists := k.GetValidatorSlashingCriteria(ctx, validator)
-				eligibleSigner := signingInfoExists && (signingStartHeight < sstxHeight)
-				deadlinePassed := sstx.Height < uint64(validator.UnbondingHeight)+params.UnbondSlashingSignerSetTxsWindow
+			for i, validator := range unbondingValidators {
+				vsi := unbondingValidatorSlashingInfos[i]
+				eligibleSigner := vsi.Exists && (vsi.SigningInfo.StartHeight < sstxHeight)
+				deadlinePassed := sstxHeight < validator.UnbondingHeight+int64(params.UnbondSlashingSignerSetTxsWindow)
```

Comment:
This adjusts from casting into `uint64` to casting into `int64`. I've also noticed when we are capturing OTX heights in two spots we are using `int64`. I know that the existing ABCI code uses that in one spot, but aren't heights all `uint64`? Why cast them into `int64` at all?

Reply:
The SDK uses int64 in some places in uint64 in others when referring to height. For example, `ValidatorSigningInfo` uses int64. I had to cast one or the other, this way had fewer casts.